src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -23,101 +23,196 @@ public DataProcessor(List<RoadAccident> roadAccidentList){
 
     /**
      * Return road accident with matching index
+     *
      * @param index
      * @return
      */
-    public RoadAccident getAccidentByIndex7(String index){
+    public RoadAccident getAccidentByIndex7(String index) {
+        Iterator<RoadAccident> roadAccidentIterator = roadAccidentList.iterator();
+        while (roadAccidentIterator.hasNext()) {
+            RoadAccident current = roadAccidentIterator.next();
+            if (current.getAccidentId().equals(index)) {
+                return current;
+            }
+        }
         return null;
     }
 
 
     /**
      * filter list by longtitude and latitude values, including boundaries
+     *
      * @param minLongitude
      * @param maxLongitude
      * @param minLatitude
      * @param maxLatitude
      * @return
      */
-    public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
+    public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude) {
+        ArrayList<RoadAccident> result = new ArrayList<>();
+
+        Iterator<RoadAccident> roadAccidentIterator = roadAccidentList.iterator();
+
+        while (roadAccidentIterator.hasNext()) {
+            RoadAccident current = roadAccidentIterator.next();
+            if (minLongitude <= current.getLongitude() && current.getLongitude() <= maxLongitude &&
```

Comment:
would be nice to extract such checks in separated methods. even if there is only two conditions if(a && b), it's still beter to extract to show the meaning. 

if(someBusinessLogicHappends(a, b) )